plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployCommandHandler.java

```diff
@@ -88,8 +86,10 @@ private IPath createWorkDirectory() throws IOException {
     return workDirectory;
   }
 
-  private Credential login(IShellProvider shellProvider) throws IOException, CoreException {
-    Credential credential = new GoogleLoginService().getActiveCredential(shellProvider);
+  private Credential login() throws CoreException {
+    IGoogleLoginService loginService =
+        PlatformUI.getWorkbench().getService(IGoogleLoginService.class);
+    Credential credential = loginService.getActiveCredential();
```

Comment:
I think thee GoogleLoginService should probably have and isLoggedIn() method and that can be called here.